a3fe/configuration/engine_config.py

```diff
@@ -5,298 +5,243 @@
+    cutoff_distance: float = _Field(12.0, alias="cutoff distance", ge=6.0, le=18.0, description="Cutoff distance in angstroms (6-18). Default 12.0 for cutoffperiodic and 10.0 for PME")
+    reaction_field_dielectric: float = _Field(78.3, alias="reaction field dielectric",
+        description="Reaction field dielectric constant (only for cutoffperiodic). "
+                    "If cutoff type is PME, this value is ignored"
+    )
+    ### Trajectory ###
+    buffered_coords_freq: int = _Field(5000,alias="buffered coordinates frequency",description="Frequency of buffered coordinates output")
+    center_solute: bool = _Field(True, alias="center solute", description="Center solute in box")
 
-        Parameters
-        ----------
-        n_cycles : int
-            Number of cycles to set in the somd config file.
-        """
-        with open(_os.path.join(self.input_dir, "somd.cfg"), "r") as ifile:
-            lines = ifile.readlines()
-        for i, line in enumerate(lines):
-            if line.startswith("ncycles ="):
-                lines[i] = "ncycles = " + str(n_cycles) + "\n"
-                break
-        #Now write the new file
-        with open(_os.path.join(self.input_dir, "somd.cfg"), "w+") as ofile:
-            for line in lines:
-                ofile.write(line)
-
-    def _validate_runtime_and_update_config(self) -> None:
-        """
-        Validate the runtime and update the simulation configuration.
+    ### Minimisation ###
+    minimise: bool = _Field(True, description="Perform energy minimisation")
 
-        Need to make sure that runtime is a multiple of the time per cycle
-        otherwise actual time could be quite different from requested runtime
+    ### Restraints ###
+    use_boresch_restraints: bool = _Field(False, description="Use Boresch restraints mode")
+    turn_on_receptor_ligand_restraints: bool = _Field(False, description="Turn on receptor-ligand restraints mode")
 
-        Raises
-        ------
-        ValueError
-            If runtime is not a multiple of the time per cycle.
-        """
-        time_per_cycle = self.timestep * self.nmoves / 1_000_000  # Convert fs to ns
-        # Convert both to float for division
-        remainder = float(self.runtime) / float(time_per_cycle)
-        if not _isclose(remainder - round(remainder), 0, abs_tol=1e-5):
+    ### Alchemistry - restraints added by a3fe ###
+    perturbed_residue_number: int = _Field(1,alias="perturbed residue number",ge=1, description="Residue number to perturb. Must be >= 1")
+    energy_frequency: int = _Field(200,alias="energy frequency",description="Frequency of energy output")
+    ligand_charge: int = _Field(0, description="Net charge of the ligand. If non-zero, must use PME for electrostatics.")
+    charge_difference: int = _Field(0, description="Charge difference between the ligand and the system. If non-zero, must use co-alchemical ion approach.")
```

Comment:
These are the same for ABFE - let's just keep the ligand charge option. We can write it out to the config file as charge_difference = - ligand charge. Thanks